src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -280,7 +287,7 @@ private async Task ProcessIncomingFramesAsync()
             {
                 if (NetEventSource.IsEnabled) Trace($"{nameof(ProcessIncomingFramesAsync)}: {e.Message}");
 
-                if (!_disposed)
+                if (_abortException == null)
```

Comment:
Why not just always call Abort and let its logic figure out what to do?

Reply:
Seems reasonable. I will make the change and see what happens.

The main point of the change here is to not skip this just because _disposed is set; this could mean we don't abort streams properly -- see my comment on _disposed at the top of the file.

FWIW, there's some weirdness here in that if the connection fails, we could end up throwing in both the read path and the write path, and racing to determine which exception is actually stored. That seems bad; better to have consistent behavior here. We should always fail in the read path, because we are constantly reading from the connection. We could also fail a write, and if that happens, that's when the race occurs. I thought about changing this so that we only actually abort when the read fails, but then what do we do if a write fails? Just keep trying to write and assume that the read will fail too, causing an abort? Seems a bit odd. Thoughts?

Reply:
Not really related, but anyway: We're also going to throw here when the server shuts down the connection and we receive EOF. We throw some exception about "expected 9 bytes to read" (frame header size). This is slightly annoying for two reasons: One is, we should avoid throwing in non-error situations, which this could be, if it the server has previously sent a GOAWAY and we've finished processing all streams. Two is, the actual exception text is not super helpful. Possibly we should make some changes here. We do need to still go through the abort path in this case because we need to abort any requests that the server didn't complete. Thoughts?

Reply:
> we could end up throwing in both the read path and the write path, and racing to determine which exception is actually stored.

I think it's ok, or at least that the alternatives are worse. But this is why we have _abortException, user an interlocked to set it, and when it's set propagate it, in order to try to always surface the same exception, even when there's a race.

Reply:
> One is, we should avoid throwing in non-error situations, which this could be, if it the server has previously sent a GOAWAY and we've finished processing all streams.

Doesn't seem particularly hard. We could just TryReadFrame or whatever, and then if it was false and there were active streams, we could abort them with a better exception.